app/services/lms_sync_service.py

```diff
@@ -91,6 +104,189 @@ async def sync_assignments(self):
                 )
         
         return canvas_assignments
+    
+    async def _pair_db_canvas_users(self, db_users: list[UserModel], canvas_users: list[dict], canvas_user_pids: list[str]):
+        # User exists in Canvas but not DB = create
+        # User exists in DB but not Canvas = delete
+        # User exists in both DB and Canvas = update
+        pairs = []
+        
+        async def process_canvas_user(canvas_user) -> ResourcePair | None:
```

Comment:
I think this should be called `create_resource_pair()` instead. "Processing" the canvas user doesn't mean anything, really.

Reply:
It's named as such because it's an internal function used to process each canvas user (as opposed to the for block underneath which "processes" each database user. I can rename it but there was intent behind the name

Reply:
To be fair, I'm not sure why I wrote it in this way. It should be directly appending rather than returning to be honest, then the name would be more suitable.

Reply:
Right but the verb "process" doesn't really indicate what's happening here. Whereas "creating" a certain data type does. I just tend not to like vague verbs in function names ya know

Reply:
The issue is the name you're suggesting is also misleading, because that function is not just creating resource pairs. It specifically is processing canvas users to derive pairs.

Reply:
Now that this method is a genuine "processor," i.e., it is processing canvas users into the pair list, I'm going to mark this resolved